Sources/SourceKitLSP/Swift/SyntaxHighlightingToken.swift

```diff
@@ -16,56 +16,48 @@ import LSPLogging
 
 /// A ranged token in the document used for syntax highlighting.
 public struct SyntaxHighlightingToken: Hashable {
-  public var start: Position
-  public var length: Int
+  /// The range of the token in the document. Must be on a single line.
+  public var range: Range<Position> {
+    didSet {
+      assert(range.lowerBound.line == range.upperBound.line)
+    }
+  }
+  /// The token type.
   public var kind: Kind
+  /// Additional metadata about the token.
   public var modifiers: Modifiers
 
-  /// The end of a token. Note that this requires the token to be
-  /// on a single line, which is the case for all tokens emitted
-  /// by parseTokens, however.
-  public var sameLineEnd: Position {
-    Position(line: start.line, utf16index: start.utf16index + length)
+  /// The (inclusive) start position of the token.
+  /// Setting it shifts the token and preserves the length.
+  public var start: Position {
+    get { range.lowerBound }
+    set {
```

Comment:
I think these setters on `start`, `end` and `utf16length` are super neat but do we use them anywhere? I couldn’t find any uses on first glance, but I might be missing something.

Reply:
Yes, in `DocumentManager.edit`: https://github.com/apple/sourcekit-lsp/pull/414/files#diff-26f557ea9c60d8fce5d02dca22daaa0e00acca75f46bd1bedc1bb1d359d73d3cR200-R206

Reply:
Ah, thanks. I missed that 👍